woodwork/accessor_utils.py

```diff
@@ -153,7 +153,8 @@ def get_invalid_schema_message(dataframe, schema):
                 f'{df_dtype}, and {schema.logical_types[name]} dtype, {valid_dtype}'
     if schema.index is not None and isinstance(dataframe, pd.DataFrame):
         # Index validation not performed for Dask/Koalas
-        if not all(dataframe.index == dataframe[schema.index]):
+        if not all((dataframe.index == dataframe[schema.index]) |
+                   (pd.isnull(dataframe.index) & pd.isnull(dataframe[schema.index]))):
```

Comment:
The other option I was considering was to fill nans and then keep the comparison the same from [here](https://stackoverflow.com/questions/19322506/pandas-dataframes-with-nans-equality-comparison).

Seems like it might be faster based on the size of the data , but not totally sure...

Reply:
I haven't tested this extensively, but I think you can also do this: 
```not pd.Series(nan_df.index).equals(pd.Series(nan_df[schema.index].values))```

Some initial test comparisons:
![image](https://user-images.githubusercontent.com/4307001/116311248-a74dc680-a770-11eb-82d9-ab7e2a2ef463.png)


This relative performance held true for 100,000 rows, 1,000,000 rows and 100,000,000 rows

Reply:
! That seems promising.

Reply:
This was as a 2 column dataframe with a single `NaN` value in the last row of the index column.

Reply:
Almost exactly the same times if I put the `NaN` value in the first row instead.

Reply:
That seems reason enough to use `.equals` here. Thanks for looking into it!

Reply:
A potential problem with using equals here is that it seems like a `pd.Index` cannot have the `string` dtype, so because it'll have the `object` dtype, the series will not be equal if the index is NaturalLanguage-like. 

```python
df = pd.DataFrame({'id':pd.Series([5,4,None,2], dtype='float64'), 'col':pd.Series(['b','k','x','d'], dtype='string')})
df.ww.init(index='col', logical_types={'col':'natural_language'})
schema = df.ww.schema

idx = df.index
col = df[schema.index]

assert df.index.astype('string').dtype == 'object'

idx_series = pd.Series(idx)
col_series = pd.Series(col.values)

assert not idx_series.equals(col_series)
```

Reply:
@thehomebrewnerd  what do you think of this modification to handle the dtype issue:
```python
pd.Series(dataframe.index, dtype=dataframe[schema.index].dtype).equals(pd.Series(dataframe[schema.index].values))
```

Reply:
Yeah, I was thinking something similar:

```
not pd.Series(nan_df.index, dtype='object').equals(pd.Series(nan_df[schema.index].values, dtype='object'))
```

Reply:
I can double check the performance of those to make sure that doesn't do something strange...

Reply:
Setting both to `object` slows it down quite a bit and makes this option the worst of the three for an integer index.

Your suggestion is still the fastest for an integer index and about a wash with the original code for a string index. The fillna approach didn't work in my test because of trying to add a new value to a 'category' dtype.

So I think your proposed solution should still be the best overall.

Reply:
This is the comparison for 10,000,000 string rows:
![image](https://user-images.githubusercontent.com/4307001/116320479-ed5d5700-a77d-11eb-956c-1624be537ace.png)

Reply:
updated to set the dtype of the index column!